### PR TITLE
First simple unit tests

### PR DESCRIPTION
**Pull request by outscale-vbr**
- net unit test
-  subnet unit test
- add in github action

### CODE REVIEW

File: controllers/osccluster_net_controller.go [hunk 1]
```diff
@@ -58,13 +56,15 @@ func reconcileNet(ctx context.Context, clusterScope *scope.ClusterScope, netsvc
 	if netSpec.ResourceId != "" {
 		netRef.ResourceMap[netName] = netSpec.ResourceId
 		netId := netSpec.ResourceId
+		clusterScope.Info("Check if the desired net exist")
```
Comment (outscale-mdr) [inline controllers/osccluster_net_controller.go:59]: Maybe add more information ? name of the net ?

File: controllers/osccluster_controller.go [hunk 2]
```diff
@@ -20,8 +20,8 @@ import (
 	"context"
 	"fmt"
 	"time"
-
 	infrastructurev1beta1 "github.com/outscale-dev/cluster-api-provider-outscale.git/api/v1beta1"
+        "github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/services/net"
```
Comment (outscale-mdr) [inline controllers/osccluster_controller.go:24]: format ?

File: controllers/osccluster_net_controller.go (hunk 1 above)
Comment (outscale-vbr) [reply controllers/osccluster_net_controller.go]: Ok done

File: controllers/osccluster_controller.go (hunk 2 above)
Comment (outscale-vbr) [reply controllers/osccluster_controller.go:24]: Ok done

**Review by outscale-mdr — APPROVED** [review]

**Comment by outscale-vbr** [comment]
close https://github.com/outscale-dev/cluster-api-provider-outscale/issues/12

**Pull request merged**